File: src/constraints/inventory_constraint.cpp
#include "inventory_constraint.h"

void InventoryConstraint::_bind_methods() {
    GDVIRTUAL_BIND(_can_add_on_inventory, "inventory", "item_id", "amount", "properties");
    GDVIRTUAL_BIND(_can_add_new_stack_on_inventory, "inventory", "item_id", "amount", "properties");
    GDVIRTUAL_BIND(_get_amount_to_add, "inventory", "item_id", "amount", "properties");
    GDVIRTUAL_BIND(_get_max_stack, "inventory", "item_id", "amount", "properties", "actual_max_stack");
}

InventoryConstraint::InventoryConstraint() {
}

InventoryConstraint::~InventoryConstraint() {
}

bool InventoryConstraint::can_add_on_inventory(const Node *inventory_node, const String item_id, const int amount, const Dictionary properties) {
	bool ret;
    if (GDVIRTUAL_CALL(_can_add_on_inventory, inventory_node, item_id, amount, properties, ret)) {
		return ret;
	}
	return true;
}

bool InventoryConstraint::can_add_new_stack_on_inventory(const Node *inventory_node, const String item_id, const int amount, const Dictionary properties) {
	bool ret;
    if (GDVIRTUAL_CALL(_can_add_new_stack_on_inventory, inventory_node, item_id, amount, properties, ret)) {
		return ret;
	}
	return true;
}

int InventoryConstraint::get_amount_to_add(const Node *inventory_node, const String item_id, const int amount, const Dictionary properties) {
	int to_add = amount;
    if (GDVIRTUAL_CALL(_get_amount_to_add, inventory_node, item_id, amount, properties, to_add)) {
		return to_add;
	}
	return to_add;
}

int InventoryConstraint::get_max_stack(const Node *inventory_node, const String item_id, const int amount, const Dictionary properties, const int actual_max_stack) {
	int new_max_stack = actual_max_stack;
	if (GDVIRTUAL_CALL(_get_max_stack, inventory_node, item_id, amount, properties, actual_max_stack, new_max_stack)) {
		return new_max_stack;
	}
	return new_max_stack;
}

File: InventorySystem_C-Sharp_Wrappers/GDExtensionWrappers/InventoryConstraint.cs
using System;
using Godot;

namespace GDExtension.Wrappers;

public partial class InventoryConstraint : Resource
{
    public static readonly StringName GDExtensionName = "InventoryConstraint";

    [Obsolete("Wrapper classes cannot be constructed with Ctor (it only instantiate the underlying Resource), please use the Instantiate() method instead.")]
    protected InventoryConstraint() { }

    /// <summary>
    /// Creates an instance of the GDExtension <see cref="InventoryConstraint"/> type, and attaches the wrapper script to it.
    /// </summary>
    /// <returns>The wrapper instance linked to the underlying GDExtension type.</returns>
    public static InventoryConstraint Instantiate()
    {
        return GDExtensionHelper.Instantiate<InventoryConstraint>(GDExtensionName);
    }

    /// <summary>
    /// Try to cast the script on the supplied <paramref name="godotObject"/> to the <see cref="InventoryConstraint"/> wrapper type,
    /// if no script has attached to the type, or the script attached to the type does not inherit the <see cref="InventoryConstraint"/> wrapper type,
    /// a new instance of the <see cref="InventoryConstraint"/> wrapper script will get attaches to the <paramref name="godotObject"/>.
    /// </summary>
    /// <remarks>The developer should only supply the <paramref name="godotObject"/> that represents the correct underlying GDExtension type.</remarks>
    /// <param name="godotObject">The <paramref name="godotObject"/> that represents the correct underlying GDExtension type.</param>
    /// <returns>The existing or a new instance of the <see cref="InventoryConstraint"/> wrapper script attached to the supplied <paramref name="godotObject"/>.</returns>
    public static InventoryConstraint Bind(GodotObject godotObject)
    {
        return GDExtensionHelper.Bind<InventoryConstraint>(godotObject);
    }
#region Methods

    public virtual bool CanAddOnInventory(Node inventory, string itemId, int amount, Godot.Collections.Dictionary properties) => Call("_can_add_on_inventory", inventory, itemId, amount, properties).As<bool>();

    public virtual bool CanAddNewStackOnInventory(Node inventory, string itemId, int amount, Godot.Collections.Dictionary properties) => Call("_can_add_new_stack_on_inventory", inventory, itemId, amount, properties).As<bool>();

    public virtual int GetAmountToAdd(Node inventory, string itemId, int amount, Godot.Collections.Dictionary properties) => Call("_get_amount_to_add", inventory, itemId, amount, properties).As<int>();

    public virtual int GetMaxStack(Node inventory, string itemId, int amount, Godot.Collections.Dictionary properties) => Call("_get_max_stack", inventory, itemId, amount, properties).As<int>();

    public virtual bool IsOverrideMaxStack(Node inventory, string itemId, int amount, Godot.Collections.Dictionary properties) => Call("_is_override_max_stack", inventory, itemId, amount, properties).As<bool>();

#endregion

}